src/portals/hope/features/claims/claim-details/ClaimCounterparty/InvoiceModal.tsx

```diff
@@ -48,31 +54,154 @@ gql`
       content
     }
   }
+
+  mutation settleRecoveryInvoice(
+    $recoveryInvoiceId: ID!
+    $request: SettleRecoveryInvoiceInput!
+  ) {
+    recoveryInvoice_settle(
+      recoveryInvoiceId: $recoveryInvoiceId
+      request: $request
+    ) {
+      id
+      isSettled
+    }
+  }
 `
 
+const ModalHeader = ({
+  title,
+  handleClose,
+}: {
+  title: string
+  handleClose: () => void
+}) => (
+  <Flex align="center" justify="space-between">
+    <h3 style={{ margin: 0 }}>{title}</h3>
+    <Button variant="tertiary" onClick={handleClose}>
+      Close
+    </Button>
+  </Flex>
+)
+
 const InvoiceFromContent: React.FC<{
   visible: boolean
   onClose: () => void
   content: string
-}> = ({ visible, onClose, content }) => {
+  invoice: RecoveryInvoice
+}> = ({ visible, onClose, content, invoice }) => {
+  const [showSettle, setShowSettle] = useState(false)
+  const [amount, setAmount] = useState(parseFloat(invoice.amount.amount))
```

Comment:
I think this works as well?
```suggestion
  const [amount, setAmount] = useState(+invoice.amount.amount)
```